exporter/exporter_test.go

```diff
@@ -59,21 +60,44 @@ func getTestExporterWithOptions(opt Options) *Exporter {
 	return e
 }
 
-func setupDBKeys(t *testing.T, uri string) error {
-	c, err := redis.DialURL(uri)
+func createClusterClient(uri string) (redis.Conn, error) {
+
+	log.Printf("Creating cluster object")
+	cluster := redisc.Cluster{
+		StartupNodes: []string{uri},
+	}
+	log.Printf("Running refresh on cluster object")
+	if err := cluster.Refresh(); err != nil {
+		log.Debugf("Cluster refresh failed: %v", err)
+		return nil, err
+	}
+
+	log.Printf("Creating redis connection object")
+	conn, err := cluster.Dial()
 	if err != nil {
-		t.Errorf("couldn't setup redis for uri %s, err: %s ", uri, err)
-		return err
+		log.Printf("Dial failed: %v", err)
+		return nil, err
 	}
-	defer c.Close()
+
+	c, err := redisc.RetryConn(conn, 10, 100*time.Millisecond)
+	if err != nil {
+		log.Printf("RetryConn failed: %v", err)
+		return nil, err
+	}
+
+	return c, nil
+
+}
+
+func setupKeys(t *testing.T, c redis.Conn, dbNumStr string) error {
 
 	if _, err := c.Do("SELECT", dbNumStr); err != nil {
 		log.Printf("setupDBKeys() - couldn't setup redis, err: %s ", err)
 		// not failing on this one - cluster doesn't allow for SELECT so we log and ignore the error
 	}
 
 	for _, key := range keys {
-		_, err = c.Do("SET", key, TestValue)
+		_, err := c.Do("SET", key, TestValue)
```

Comment:
Good catch. 
You could even change this to `if _, err := c.Do("SET", key, TestValue); err != nil { ` to make sure the scoping is as tight as possible.